middleware.go

```diff
@@ -80,9 +86,14 @@ func createMiddleware(mw TykMiddleware) func(http.Handler) http.Handler {
 				h.ServeHTTP(w, r)
 				return
 			}
+
 			err, errCode := mw.ProcessRequest(w, r, mwConf)
 			if err != nil {
 
+				if errCode == http.StatusForbidden {
```

Comment:
how would we know that this `http.StatusForbidden` caused by one of our middlewares, not from actual upstream?
I think if it is reply from upstream we don't need to do anything as it was resposibility of upstream to set CORS headers and they might already there.

also, how about other HTTP 4XX which can be set by other our middlewares?

Reply:
Because at the point this has been injected - it's from the middleware, and not the upstream. So at this stage, the request has not yet been proxied and we are about to return an error to the user.

Reply:
my understanding is that this depends on order of middlewares. it is not super clear for me that  the request has not yet been proxied as this code will be literally part of every middleware (if I understand right)